toolkit/tools/internal/retry/retry.go

```diff
@@ -4,13 +4,29 @@
 package retry
 
 import (
+	"math"
 	"time"
 )
 
+func calculateDelay(retryCount int, sleep time.Duration, backoffExponent float64) time.Duration {
+	// Calculate an exponential backoff.
+	// The formula is: sleep * (backoffExp ^ retryCount)
+	// For example, if sleep = 1 second, backoffExp = 2.0, retryCount = 3
+	// then the delay will be 1 * (2 ^ 3) = 8 seconds.
+	expRetry := math.Pow(backoffExp, float64(retryCount))
```

Comment:
nit: `backoffExponent` (name wasn't updated)

Reply:
Given the change in fuction, working to rewrite a bit of stuff. Change is included in that (I just accepted the GitHub suggestions for a bunch of these, so only part of it is available online right now)